docs/spec/http_response.json

```diff
@@ -0,0 +1,36 @@
+{
+    "$id": "doc/spec/http_response.json",
+    "title": "HTTP response object",
+    "description": "HTTP response object, used by error, span and transction documents",
+    "type": ["object", "null"],
+    "properties": {
+        "status_code": {
+            "type": ["integer", "null"],
+            "description": "The status code of the http request."
+        },
+        "transfer_size": {
+            "type": ["number", "null"],
+            "description": "Total size of the payload."
+        },
+        "encoded_body_size": {
+            "type": ["number", "null"],
+            "description": "The encoded size of the payload."
+        },
+        "decoded_body_size":  {
+            "type": ["number", "null"],
+            "description": "The decoded size of the payload."
+        },
+        "headers": {
+            "type": ["object", "null"],
+            "patternProperties": {
+                "[.*]*$": {
+                    "type": ["string", "array", "null"],
+                    "items": {
+                        "type": ["string"]
+                    }
+                }
+            },
+            "additionalProperties": false
```

Comment:
Since the regex for the pattern property matches against everything, there is no logical change. The intention was to make it clear that only those defined headers are allowed.